src/applications/personalization/dashboard/components/benefit-application-drafts/SubmissionCard.jsx

```diff
@@ -25,6 +29,110 @@ ReceivedContent.propTypes = {
   lastSavedDate: PropTypes.string.isRequired,
 };
 
+/**
+ * todo:
+ *
+ * Date calculation
+ * - pass in received date
+ * - add logic for expiration (60 days)
+ */
+const SavePdfDownload = ({
+  formId,
+  getPdfDownloadUrl,
+  guid,
+  showLoadingIndicator,
+}) => {
+  const [showSuccess, setShowSuccess] = useState(false);
+  const [error, setError] = useState('');
+  const [showDownloadingButton, setShowDownloadingButton] = useState(false);
+
+  useEffect(
+    () => {
+      if (!showLoadingIndicator) {
+        setTimeout(() => setShowDownloadingButton(false), 3000);
+      }
+    },
+    [showLoadingIndicator],
+  );
+
+  const handleDownloadButtonClick = async () => {
+    setError(null);
+    setShowDownloadingButton(true);
+    const result = await getPdfDownloadUrl(formId, guid);
+
+    if (result.error) {
+      setError(result.error);
+    } else {
+      setShowSuccess(true);
+      // setShowDownloadingButton(false);
```

Comment:
did we want to keep this comment?
and out of curiosity: will this button be disabled while the user waits for the download?

Reply:
Comment removed.

Because the button is asking the API to generate the PDF for download we are disabling the button while it's generating and then for 3 seconds after the file download starts.